File: Data Structures & Algorithms/subtree-of-a-binary-tree/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {  
    public boolean isSubtree(TreeNode root, TreeNode subRoot) {
        Queue<TreeNode> tracker = new LinkedList();

        tracker.add(root); 
        while (!tracker.isEmpty()){
            TreeNode curr = tracker.peek();
            if (curr.val == subRoot.val){
                if (isSubtreeHelper(curr, subRoot)){
                    return true;
                }
            }
            tracker.remove();
            if (curr.left != null) tracker.add(curr.left);
            if (curr.right != null) tracker.add(curr.right);
        }
        return false;
    }

    private boolean isSubtreeHelper(TreeNode root, TreeNode subRoot){
        if (root== null && subRoot == null) {
            return root == subRoot;
        } else if (root != null && subRoot != null){
            if (root.val == subRoot.val){
                return isSubtreeHelper(root.left, subRoot.left) && isSubtreeHelper(root.right, subRoot.right);
            } else {
                return false;
            }
        } else {
                return false;
            
        }
    }
}

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        res = [] #create an array for the result

        q = collections.deque()#create a queue called q
        q.append(root)#add the root to the queue

        while q: #while q isnt empty
            qLen = len(q) #length of q
            level = []#create level array

            for i in range(qLen):#while i is less than current q
                node = q.popleft()
                if node:
                    level.append(node.val)
                    q.append(node.left)
                    q.append(node.right)
                
            if level:
                res.append(level)

        return res






        